### PR TITLE
FIX Ensure only live content is indexed via getMainContent()

### PR DESCRIPTION
**Pull request by josephlewisnz**

### CODE REVIEW

**Comment by josephlewisnz** [comment]
Related #64

File: src/Service/AlgoliaPageCrawler.php [hunk 1]
```diff
@@ -64,6 +65,10 @@ public function getMainContent(): string
             $useXpath = false;
             $selector = $this->config()->get('content_element_tag');
         }
+        
+        $originalStage = Versioned::get_stage();
+        //Always set to live to ensure we don't pick up draft content in our render eg. draft elemental blocks
+        Versioned::set_stage('Live');
```
Comment (wilr) [inline src/Service/AlgoliaPageCrawler.php:71]: @josephlewisnz in these cases could you please use `Versioned::set_stage(Versioned::LIVE)` just prevents any issues.

Comment (josephlewisnz) [reply src/Service/AlgoliaPageCrawler.php]: I've added that  now - I'll do some additional testing today

**Comment by josephlewisnz** [comment]
https://github.com/wilr/silverstripe-algolia/issues/64#issuecomment-1720185354

**Comment by wilr** [comment]
@josephlewisnz Thanks! Definitely will add a test for this as soon as possible as well

**Comment by kinglozzer** [comment]
Any chance of a tag for this? 👼

**Pull request merged**